diff -Naur frei0r-20100226-org/src/filter/curves/curves.c frei0r-20100226/src/filter/curves/curves.c
--- frei0r-20100226-org/src/filter/curves/curves.c	2010-02-26 06:59:10.000000000 +0100
+++ frei0r-20100226/src/filter/curves/curves.c	2010-02-27 14:02:41.000000000 +0100
@@ -449,9 +449,9 @@
 		for(int j = 0; j < scale; j++) {
 		  if (j % cellSize > lineWidth) { //point doesn't aly on the grid
 			int offset = ((maxYvalue - i + graphYOffset) * stride + j + graphXOffset) * 4;
-			dst[offset] = (dst[offset++] >> 1) + 0x7F;
-			dst[offset] = (dst[offset++] >> 1) + 0x7F;
-			dst[offset] = (dst[offset++] >> 1) + 0x7F;
+			dst[offset] = (dst[offset] >> 1) + 0x7F; offset++;
+			dst[offset] = (dst[offset] >> 1) + 0x7F; offset++;
+			dst[offset] = (dst[offset] >> 1) + 0x7F; offset++;
 		  }
 		}
 	}
diff -Naur frei0r-20100226-org/src/filter/three_point_balance/three_point_balance.c frei0r-20100226/src/filter/three_point_balance/three_point_balance.c
--- frei0r-20100226-org/src/filter/three_point_balance/three_point_balance.c	2010-02-26 06:59:10.000000000 +0100
+++ frei0r-20100226/src/filter/three_point_balance/three_point_balance.c	2010-02-27 14:14:08.000000000 +0100
@@ -263,15 +263,15 @@
 	for(int i = 0; i < inst->height; i++) {
 	  int offset = (i * inst->width + j) * 4;
 	  if (copyPixel) {
-		dst[offset] = src[offset++];
-		dst[offset] = src[offset++];
-		dst[offset] = src[offset++];
+		dst[offset] = src[offset]; offset++;
+		dst[offset] = src[offset]; offset++;
+		dst[offset] = src[offset]; offset++;
 	  } else {
-		dst[offset] = mapRed[src[offset++]];
-		dst[offset] = mapGreen[src[offset++]];
-		dst[offset] = mapBlue[src[offset++]];
+		dst[offset] = mapRed[src[offset]]; offset++;
+		dst[offset] = mapGreen[src[offset]]; offset++;
+		dst[offset] = mapBlue[src[offset]]; offset++;
 	  }
-	  dst[offset] = src[offset++]; // copy alpha
+	  dst[offset] = src[offset]; offset++;// copy alpha
 	}
   }
   
