diff -Naur frei0r-20100226-org/src/filter/curves/curves.c frei0r-20100226/src/filter/curves/curves.c
--- frei0r-20100226-org/src/filter/curves/curves.c	2010-02-26 06:59:10.000000000 +0100
+++ frei0r-20100226/src/filter/curves/curves.c	2010-02-26 18:48:08.000000000 +0100
@@ -334,6 +334,7 @@
 		double dx = x - coeffs[offset];
 		return ((coeffs[offset + 4] * dx / 6. + coeffs[offset + 3] / 2.) * dx + coeffs[offset + 2]) * dx + coeffs[offset + 1];
 	}
+	return 0.;
 }
 
 void swap(double *points, int i, int j) {
